docs/DeveloperGuide.md

```diff
@@ -1,34 +1,182 @@
+*Figure #. Interactions Inside the Logic Component for the `clear` Command*
+
+### Model component
+
+**API** : [`seedu.duke.person`](https://github.com/AY2021S2-CS2113T-T09-1/tp/tree/master/src/main/java/seedu/duke/person) package
+
+
+<img src="images/ModelComponentStructure.png" width="600" height="600" />
+
+*Figure #. Structure of the Model Component*
+
+The Model component,
+
+* contains a `Person` class which represents a person who checks in/out.
+* contains a `TrackingList` class which uses an ArrayList to keep track of all the `Person` objects who have currently checked in/out.
+* contains a `PersonLog` class which uses a HashMap to permanently store all the `Person` objects who have checked in before.
+
+A `Person` object contains:
+* an `Id` object
+* a `Name` object
+* a `Phone` object
+
+
+### Storage component
+**API** : [`seedu.duke.storage`](https://github.com/AY2021S2-CS2113T-T09-1/tp/tree/master/src/main/java/seedu/duke/storage) package
+![](images/storage_module.png?raw=true "Storage Module diagram")
+
+The `Storage` component,
+* saves and encodes `Person` objects in `Tracking List` into a `.txt` file.
+* reads a `.txt` file of valid encoding and populates a `TrackingList`
+* records a log of all historical states of `TrackingList`
+* keeps a list of user setting in a `Config.properties` file which is read on launch
+* defaults to `TrackingList.txt` but can be saved at a custom specified location
+* loads on program startup automatically saves after each command
+
+The storage module uses the `StorageFile` class to store the `TrackingList` class from Model. 
+`StorageFile` achieves this by using two helper classes, `TrackingListEncoder` and `TrackingListDecoder`.
+
+`TrackingListEncoder` takes the currently stored `TrackingList` and converts it to a more storage friendly String.
+This String is then written to a file on the disk, known on the diagram as `StoredTrackingList`.
+As the name suggests, `TrackingListDecoder` achieves the reverse by taking the file on drive, `StoredTrackingList`,
+and interprets the file to be loaded into the memory as a `TrackingList`.
+
+`StoredTrackingList` is the `.txt` file stored in the same directory as the program.
+The location is determined by the `ConfigFile` class. 
+
+`LogFile` takes the `TrackingList` every time the storage is updated
+and packages it into `json` formatted string.
+`LogFile` communicates with the disk by calling the methods
+`loadLogFile` and `saveLogFile` in the `StorageFile` class.
+
+
+The program settings for the user is manged by the `ConfigFile` class. 
+Another file, `settingsFile` is read by `ConfigFile` when the program is started, 
+to load the settings into the program memory. 
+When a change is made by the user, `ConfigFile` will update the `settingsFile` through a save function.
+
+### Common classes
+
+Classes used by multiple components are in the `seedu.Duke.commons` package. 
+
+## Implementation
```

Comment:
The implementation is quite important in DG. Perhaps your team hasn't started yet, it would be better to show every important feature here with the help of literal explanation and sequence diagrams! Looking forward to your results.